### PR TITLE
feat: adding default algorand network configs to use when no .env.{network} found

### PR DESCRIPTION
**Pull request by aorumbayev**
https://github.com/algorandfoundation/algokit-cli/issues/532

## Proposed changes

Adding defaults to use if user is invoking algokit project deploy {mainnet|localnet|testnet} and no .env.{network_name} exists. This should not prevent user from deploying with defaults (and in case of testnet|mainnet it will use free public infrastructure - algonode). But the main reason is to reduce noisy amount of .env files in smart contract based algokit templates.

### CODE REVIEW

File: src/algokit/core/project/deploy.py [hunk 1]
```diff
@@ -4,13 +4,55 @@
...
+    _KnownEnvironments.LOCALNET: {
+        # this file should contain environment variables specific to algokit localnet
+        "ALGOD_TOKEN": str(DEFAULT_ALGOD_TOKEN),
+        "ALGOD_SERVER": str(DEFAULT_ALGOD_SERVER),
+        "ALGOD_PORT": str(DEFAULT_ALGOD_PORT),
+        "INDEXER_TOKEN": str(DEFAULT_INDEXER_TOKEN),
+        "INDEXER_SERVER": str(DEFAULT_INDEXER_SERVER),
+        "INDEXER_PORT": str(DEFAULT_INDEXER_PORT),
+    },
+    _KnownEnvironments.MAINNET: {
```
Comment (neilcampbell) [inline src/algokit/core/project/deploy.py:45]: Do we need ports for the mainnet and testnet configs?

File: src/algokit/core/project/deploy.py [hunk 2]
```diff
@@ -4,13 +4,55 @@
...
 
+class _KnownEnvironments:
+    LOCALNET = "localnet"
+    MAINNET = "mainnet"
+    TESTNET = "testnet"
+
+
+_mainnet_algod_config = get_algonode_config("mainnet", config="algod", token=DEFAULT_ALGOD_TOKEN)
+_testnet_algod_config = get_algonode_config("testnet", config="algod", token=DEFAULT_ALGOD_TOKEN)
+_mainnet_indexer_config = get_algonode_config("mainnet", config="indexer", token=DEFAULT_INDEXER_TOKEN)
```
Comment (neilcampbell) [inline src/algokit/core/project/deploy.py:31]: AFAIK no token is needed for algonode.

Comment (robdmoore) [reply src/algokit/core/project/deploy.py:31]: +1

File: src/algokit/core/project/deploy.py (hunk 1 above)
Comment (robdmoore) [reply src/algokit/core/project/deploy.py:45]: +1 - should be 443 right?

Comment (aorumbayev) [reply src/algokit/core/project/deploy.py:45]: @robdmoore @neilcampbell i used the values from localnet config from .env.template files we got on smart contract templates. The values are as follows:

ALGOD_TOKEN=aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa
ALGOD_SERVER=http://localhost
ALGOD_PORT=4001
INDEXER_TOKEN=aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa
INDEXER_SERVER=http://localhost
INDEXER_PORT=8980

I am pretty sure ports are needed but only on localnet, there is a reason why we got it this way on the template repos. Will double check the behaviour where ports are utilized specifically in localnet cases, i think there is some methods in algokit-utils that rely on existence of those vars in localnet scenarios.

**Pull request merged**